docs/CodingGuidelines.md

```diff
@@ -0,0 +1,184 @@
+    ```  
+
+14. Make the code consistent with the existing code base
+15. Avoid mutations when possible. Prefer immutability
+    *GOOD EXAMPLE*:  
+    Returns updated object without mutating the input
+     ```  
+     (myObject) => ({ ...myObject, name: trimName(myObject.name (http://myobject.name/)) }) 
+    ```  
+    *BAD EXAMPLE*:  
+    Mutates input to method (DO NOT DO THIS)  
+    `(myObject) => { myObject.name (http://myobject.name/) = trimName(myObject.name) } `
+16. Keep it simple
+
+    Code bases tend to only grow in complexity, until no one can understand them or maintain them. It is paramount that reducing complexity is a top concern on every piece of code.
+
+17. Write functions as pure functions where possible.
+
+    A pure function is a function which takes inputs, and returns a result, without modifying the inputs and without any side effects.
+
+    *EXAMPLE OF PURE FUNCTION*:  
+    
+    `(a, b) => a + b`
+
+    *EXAMPLE OF IMPURE FUNCTION*:  
+    
+    `() => { globalCounter++; }`
+    
+    Impure functions are unavoidable, but should be minimized, and should contain the minimal required. Complicated business logic should be separated off into independent pure functions. Pure functions are much easier to test, and utilize.
+
+18. Consider performance where it matters
+
+    In some scenarios, improving the performance will add complexity, while not actually effecting the end performance seen by the users. In this case, always opt for simplicity.
+
+19. Embrace refactoring
+
+    Refactoring is crucial to the health of any code base. Fearlessly refactor where it can help improve simplicity. Often times code when originally written misses the mark. Do not be afraid to refactor code written by other authors.
+20. State is the root of complexity
+
+    Minimize the amount of state is required and avoid entangling state management through the entire codebase.  Often times state should be pushed towards the boundary (such as in a reducer).
+
+    The being said there’s nothing wrong with local UI state managed directly in a component. For example, for a collapsable menu, it will probably make sense to just store the open state directly in the component. Do not blindly pull all state into a centralized state manager as a reducer.
+
+    *EXAMPLE OF BAD STATE MANAGEMENT*:
+
+    UI components storing state of multiple async calls, coordinating loading status, and pagination
+
+## IoT App Kit component requirements
+
+1. Components will utilize [Cloudscape](https://cloudscape.design/) for primitive components, unless a good case is made not to. The default is to use Cloudscape.
+1. Styling should adhere to https://cloudscape.design/ design principles and guidelines.
+1. Web components will be exported within https://github.com/awslabs/iot-app-kit/tree/main/packages/components, with a iot- prefix. i.e. iot-line-chart. Their styles should be contained within `"@iot-app-kit/components/styles.css".
```

Comment:
You're missing a final backtick at the end of this line, so it doesn't come out as a code block